src/c2pa/c2pa.py

```diff
@@ -679,6 +679,67 @@ def sign_file(
             signer.close()
 
 
+def sign_file_with_callback_signer(
+    source_path: Union[str, Path],
+    dest_path: Union[str, Path],
+    manifest: str,
+    signer: 'Signer'
```

Comment:
For this one, the Signer should be created outside the function, so the caller totally handles and owns the callback.